scripts/start

```diff
@@ -0,0 +1,159 @@
+  # server side
+  checkRequirement java
+  checkRequirement sbt
+
+  # client side
+  checkRequirement npm
+
+  # used for postgres and AWS localstack
+  checkRequirement docker
+
+  # other
+  checkRequirement nginx
+  checkRequirement aws
+}
+
+tunnelToAwsDb() {
+    if (docker stats --no-stream &> /dev/null); then
+      docker compose down
+    fi
+    if [[ -n $EXISTING_TUNNELS ]]; then
+      echo "RE-USING EXISTING TUNNEL TO CODE POSTGRES (on port ${POSTGRES_PORT})"
+    else
+      # todo: Grid script uses the following options, do we need any of these?
+      #  TUNNEL_OPTS="-o ExitOnForwardFailure=yes -o ServerAliveInterval=10 -o ServerAliveCountMax=2"
+      ssm ssh -t ${APP_NAME},CODE -p ${AWS_PROFILE} -x --newest --rds-tunnel 5432:${APP_NAME},CODE
+      echo "TUNNEL ESTABLISHED TO CODE POSTGRES (on port ${POSTGRES_PORT})"
+    fi
+}
+
+startDockerContainers() {
+  if [[ $EXISTING_TUNNELS ]]; then
+    echo "KILLING EXISTING TUNNEL TO CODE POSTGRES (on port ${POSTGRES_PORT})"
+    # shellcheck disable=SC2046
+    kill $(echo "${EXISTING_TUNNELS}" | awk '{print $2}')
+  fi
+  docker compose up -d
+}
+
+startIngestionLambda() {
+  pushd "$ROOT_DIR/ingestion-lambda"
+  npm install
+  npm run dev
+  popd
+}
+
+startPlayApp() {
+  # pushd to find build.sbt and allow this script to be executed from any location (but ideally from the project root)
+  echo "========================================================="
+  echo "= Press cmd-C then cmd-D to stop"
+  echo "========================================================="
+  pushd "$ROOT_DIR/${APP_NAME}"
+  if [ "$IS_DEBUG" == true ] ; then
+    SBT_OPTS="-jvm-debug 5005"
+  fi
+  if [[ $USE_CODE == true ]]; then
+    sbt "${SBT_OPTS} run"
+  else
+    sbt "${SBT_OPTS} run"
+  fi
+  popd
+}
+
+checkNodeVersion() {
+  runningNodeVersion=$(node -v)
+  requiredNodeVersion=$(cat "$ROOT_DIR/.nvmrc")
+
+  if [ "$runningNodeVersion" != "$requiredNodeVersion" ]; then
+    echo -e "${red}Using wrong version of Node. Required ${requiredNodeVersion}. Running ${runningNodeVersion}.${plain}"
+    exit 1
+  fi
+}
+
+checkForJavaHome() {
+  echo "Checking JAVA_HOME"
+  if [[ -z "$JAVA_HOME" ]]; then
+    echo "  JAVA_HOME not set, please set it before continuing"
+    echo "  This can be done by adding \"export JAVA_HOME=\$(/usr/libexec/java_home)\" to ~/.profile"
+    exit 1
+  else
+    echo "  JAVA_HOME is set to $JAVA_HOME"
+  fi
+}
+
+main() {
+    checkForJavaHome
+    hasCredentials
+    checkRequirements
+    checkNodeVersion
+    if [ $USE_CODE == true ]; then
+        echo "Using CODE. Starting play app with tunnel to AWS DB."
+        tunnelToAwsDb && USE_CODE_DB=true startPlayApp
+    else
+      echo "Using the local stack. Starting Docker, Ingestion Lambda and Play App."
+      # run all three, and terminate together when the user presses `ctrl+c`: https://stackoverflow.com/a/52033580
+      (trap 'kill 0' SIGINT; startDockerContainers && startIngestionLambda & startPlayApp & wait)
```

Comment:
this seems to broadly work as expected, but I've noticed a couple of bugs with my shell when subsequently doing interactive stuff in the same shell session, so I think this needs a bit more investigation..

Reply:
I've had trouble with backgrounding via `&` before, let's see how painful it gets when using day-to-day and improve in a future PR if needs be